### DISCUSSION TITLE
Is it possilbe to turn of loggin whe using GraphServiceClient in azure function?

### DISCUSSION THREAD

**Discussion by PanClifton** [microsoftgraph/msgraph-sdk-dotnet#2770] [Q&A] [answered]
hello 
I am using GraphServiceClient in my project.

here is initialisation of GraphServiceClient 
```
public sealed class GraphClientFactory : IGraphClientFactory
{
    public GraphServiceClient GraphServiceClientInstance { get; }
    public ServicePrincipalItemRequestBuilder HeiDIServicePrincipalItemRequestBuilder { get; }

    public GraphClientFactory(GraphConfiguration graphConfiguration)
    {
        var clientSecretCredential = new ClientSecretCredential(
                graphConfiguration.TenantId, graphConfiguration.ClientId, graphConfiguration.ClientSecret);

        GraphServiceClientInstance = new GraphServiceClient(clientSecretCredential);
        HeiDIServicePrincipalItemRequestBuilder = GraphServiceClientInstance.ServicePrincipals[graphConfiguration.ServicePrincipalId];
    }
}
```

i am getting loads of logging  from GraphServiceClient in azure function console.
is it possible to turn of logging from inside the GraphServiceClient?

**Comment by andrueastman** [comment]
Thanks for raising this @PanClifton 

Any chance you can confirm the LogLevel and Category you are seeing in the logs generated? I suspect they may be from the HttpClient that the is used internally.

**Reply by PanClifton** [reply]
Hi here is my host.json file

```
{
  "version": "2.0",
  "logging": {
    "applicationInsights": {
      "samplingSettings": {
        "isEnabled": true,
        "excludedTypes": "Request"
      },
      "enableLiveMetricsFilters": true
    },
    "logLevel": {
      "default": "Information",
      "Azure.Core": "Warning"
    }
  }
}
```
btw i am using 
```
<PackageReference Include="Microsoft.Graph" Version="5.65.0" />
```

**Reply by PanClifton** [reply]
this is sample output of azure function console
![image](https://github.com/user-attachments/assets/5faafe13-2692-4b89-9a8a-52ae35e2042c)

**Reply by pschaeflein** [reply, marked as answer]
Those entries are from the MSAL library. 

More detail is here: https://learn.microsoft.com/en-us/entra/msal/dotnet/advanced/exceptions/msal-logging?tabs=dotnet

**Reply by PanClifton** [reply]
Thanks for the answers. it helped
